JavaCryptographicArchitecture/src/BC/KeyGenerator.cryptsl

```diff
@@ -0,0 +1,39 @@
+SPEC javax.crypto.KeyGenerator
+OBJECTS
+    int secretKeySize;
+    java.security.spec.AlgorithmParameterSpec params;
+    javax.crypto.SecretKey key;
+    java.lang.String secretKeyAlgorithm;
+    java.security.SecureRandom ranGen;
+
+EVENTS
+    g1: getInstance(secretKeyAlgorithm);
+    g2: getInstance(secretKeyAlgorithm, _);
+    Gets := g1 | g2;
+
+    i1: init(secretKeySize);
+    i2: init(secretKeySize, ranGen);
+    i3: init(params);
+    i4: init(params, ranGen);
+    i5: init(ranGen);
+    Inits := i1 | i2 | i3 | i4 | i5;
+    
+    gk: key = generateKey();
+
+ORDER
+    Gets, Inits?, gk
+
+CONSTRAINTS
+	secretKeyAlgorithm in {"AES", "HmacSHA224", "HmacSHA256", "HmacSHA384", "HmacSHA512", "Poly1305-AES", "Blowfish", "Poly1305-Camellia", "ChaCha", 
+							"ChaCha7539", "DESede", "Poly1305", "Rijndael", "Salsa20", "Serpent", "Tnepres", "Poly1305-Serpent", "SHACAL-2", "Twofish", 
+							"Poly1305-Twofish", "XSalsa20"};
+    secretKeyAlgorithm in {"AES", "Poly1305-Camellia"} => secretKeySize in {128, 192, 256};
+    secretKeyAlgorithm in {"Blowfish", "ChaCha", "Salsa20", "SHACAL-2"} => secretKeySize in {128};
+    secretKeyAlgorithm in {"DESede", "Rijndael", "Serpent", "Tnepres"} => secretKeySize in {192};
```

Comment:
In theory, Rijndael should support the same key sizes as AES. In practice it also does. But when I was checking the code base of BC provider for the above algorithms, I saw that they had used those key sizes for each respective algorithm in their constructor calls. Because of that, I thought that is the secure key size that should be used for each respective algorithm. Below are the links for each algorithm where I based my conclusion.

[Rijndael](https://github.com/bcgit/bc-java/blob/master/prov/src/main/java/org/bouncycastle/jcajce/provider/symmetric/Rijndael.java#L39)
[ChaCha](https://github.com/bcgit/bc-java/blob/master/prov/src/main/java/org/bouncycastle/jcajce/provider/symmetric/ChaCha.java#L35)
[Salsa20](https://github.com/bcgit/bc-java/blob/master/prov/src/main/java/org/bouncycastle/jcajce/provider/symmetric/Salsa20.java#L31)